addons/account/tests/test_account_move_out_invoice.py

```diff
@@ -4082,3 +4082,36 @@ def test_out_invoice_create_cross_branch_refund(self):
             move_reversal.refund_moves()
 
         self.assertEqual(error_catcher.exception.args[0], "All selected moves for reversal must belong to the same company.")
+
+    def test_invoice_copy_data(self):
+        """User should be able to duplicate invoices with different journals"""
+
+        new_sale_journal = self.company_data['default_journal_sale'].copy()
+
+        invoice_1 = self.env['account.move'].create({
```

Comment:
You can create in batch (and no need to define journal_ids to reproduce the issue)
```diff
         new_sale_journal = self.company_data['default_journal_sale'].copy()
 
-        invoice_1 = self.env['account.move'].create({
+        invoice_1, invoice_2 = self.env['account.move'].create([{
             'move_type': 'out_invoice',
             'partner_id': self.partner_a.id,
             'journal_id': self.company_data['default_journal_sale'].id,
-            'invoice_line_ids': [(0, 0, {
-                'name': 'product 1',
-                'price_unit': 10,
-                'quantity': 1,
-            })],
-        })
-
-        invoice_2 = self.env['account.move'].create({
+        }, {
             'move_type': 'out_invoice',
             'partner_id': self.partner_a.id,
             'journal_id': new_sale_journal.id,
-            'invoice_line_ids': [(0, 0, {
-                'name': 'product 2',
-                'price_unit': 20,
-                'quantity': 2,
-            })],
-        })
-
-        invoices = invoice_1 + invoice_2
-        invoices_duplicate = invoices.copy_data()
+        }])
 
+        invoices_duplicate = (invoice_1 + invoice_2).copy_data()
```

Reply:
Hello @yosa-odoo :wave: 
Since this commit https://github.com/odoo/odoo/commit/4ac2702c31f0e95f33f9ad554e7350bef9dab8bd invoices will duplicate in batch
so, in the earlier versions issue is not present.
requested changes are done.
Thanks!

Reply:
The test you created fails in 16, the change is not big; I would say why not apply the fix as low (in version) as possible?

Reply:
Hello @yosa-odoo :wave: 
Yes, test case fails in 16 because we can not duplicate multiple invoices at a time in 16.0
that's why that code doesn't throw an error in 16.
and regarding the fix, issue was not producible in the earlier version, that's why I applied the fix in 17.2
Thanks! :slightly_smiling_face:

Reply:
Yep my bad!